docs/wallet_browser_extension/provider/api_documentation.md

```diff
@@ -0,0 +1,542 @@
+}
+
+interface TransactionSummary {
+  amount: string;
+  block_hash: string;
+  commands: string[];
+  command: string;
+  exit_status?: string;
+  gas_consumed: string;
+  gas_limit: string;
+  hash: string;
+  nonce: string;
+  number: string;
+  signer: string;
+  status: number;
+  timestamp: number;
+  tx_gas_consumption: string;
+}
+type GetActiveAcconutLatestTransactionsResult = TransactionSummary[];
+```
+
+=== "RPC"
+    ```js
+    window.xpll.request({
+      method: "get_active_account_latest_transactions",
+      params: {
+        limit: 1,
+        order: "desc",
+      },
+    });
+    ```
+
+=== "Helper Function"
+    ```js
+    window.xpll.getActiveAcconutLatestTransactions({
+      limit: 1,
+      order: "desc",
+    });
+    ```
+
+## Get Current Network
+
+Returns the current network.
+
+```ts
+interface GetCurrentNetworkRequest {
+  method: "get_current_network";
+}
+type GetCurrentNetworkResult = string | undefined;
+```
+
+=== "RPC"
+    ```js
+    window.xpll.request({
+      method: "get_current_network";
+    });
+    ```
+
+=== "Helper Function"
+    ```js
+    window.xpll.getCurrentNetwork();
+    ```
+
+## Send Token
+
+Transfer tokens to another account, return the transaction hash if the request is successful.
+
+### Params
+
+- `amount`: The amount of tokens to transfer in the smallest unit (e.g. [Gray][5]) of the token.
+- `recipient`: The address of the recipient account.
+- `nonce`: The nonce of the transaction.
+  If not provided, the nonce will be calculated automatically.
+- `gas_limit`: The gas limit of the transaction.
+  If not provided, the gas limit will be calculated automatically.
+- `max_base_fee_per_gas`: The maximum base fee per gas of the transaction. The default value is `8`.
+- `priority_fee_per_gas`: The priority fee per gas of the transaction. The default value is `0`.
+
+!!! warning
+    The request causes a popup to appear.
+    You should only request permissions in response to a direct user action, such as a button click.
+
+!!! failure
+    If user reject the request, the `User Rejected Request 4001` error will be thrown.
+
+```ts
+interface SendTokenRequest {
+  method: "send_token";
+  params: {
+    amount: string;
+    recipient: string;
+    nonce?: string;
+    gas_limit?: string;
+    max_base_fee_per_gas?: string;
+    priority_fee_per_gas?: string;
+  };
+}
+export type SendTokenResult = string; // transaction hash
+```
+
+=== "RPC"
+    ```js
+    window.xpll.request({
+      method: "send_token",
+      params: {
+        recipient: "FOg6_UOmsPRs4KOECp3G2UoSS1sUQQH8NSgNa_IkQ_8",
+        amount: "1",
+        gas_limit: "166350",
+        nonce: "17",
+        max_base_fee_per_gas: "8",
+        priority_fee_per_gas: "0",
+      },
+    });
+    ```
+
+=== "Helper Function"
+    ```js
+    window.xpll.sendToken({
+      recipient: "FOg6_UOmsPRs4KOECp3G2UoSS1sUQQH8NSgNa_IkQ_8",
+      amount: "1",
+    });
+    ```
+
+## Call Contract
+
+Transfer tokens to another account,
```

Comment:
Is the call contract in the extension only for transferring tokens?

Reply:
It works for the method in the smart contract except for the view method.
It seems the description is misleading. I will update it soon.

Reply:
It works for the method in the smart contract except for the view method.
It seems the description is misleading. I will update it soon.

Reply:
solved by 9f9bc2d